Tests/SentryTests/SentryTracerTests.swift

```diff
@@ -0,0 +1,56 @@
+import XCTest
+
+class SentryTracerTests: XCTestCase {
+    private class Fixture {
+        
+        let transactionName = "Some Transaction"
+        let transactionOperation = "Some Operation"
+        var transactionContext: TransactionContext!
+        
+        init() {
+            transactionContext = TransactionContext(name: transactionName, operation: transactionOperation)
+        }
+        
+        func getSut(waitForChildren: Bool = true) -> SentryTracer {
+            return SentryTracer(transactionContext: transactionContext, hub: nil, waitForChildren: waitForChildren)
+        }
+        
+    }
+    
+    private var fixture: Fixture!
+    
+    override func setUp() {
+        fixture = Fixture()
+    }
+    
+    func testSpanHierarchy() {
+        let tracer = fixture.getSut()
+        let child = tracer.startChild(operation: fixture.transactionOperation)
+        let grandChild = child.startChild(operation: fixture.transactionOperation)
+        
+        let tracerChildren = Dynamic(tracer).spans.asArray!
+        let childChildren = Dynamic(child).spans.asArray!
+        
+        XCTAssertTrue(tracerChildren.contains(child))
+        XCTAssertTrue(childChildren.contains(grandChild))
+        XCTAssertFalse(tracerChildren.contains(grandChild))
```

Comment:
`l`: Also checking the size of the array makes the test more precise.
```suggestion
        let tracerSpans = Dynamic(tracer).spans.asArray!
        let childSpan = Dynamic(child).spans.asArray!
        
        XCTAssertTrue(tracerSpans.contains(child))
        XCTAssertTrue(childSpan.contains(grandChild))
        XCTAssertFalse(tracerSpans.contains(grandChild))
        
        XCTAssertEqual(tracerSpans.count, 1)
        XCTAssertEqual(childSpan.count, 1)
```